diff -Naur URPM-4.48.2.orig/URPM/Resolve.pm URPM-4.48.2/URPM/Resolve.pm
--- URPM-4.48.2.orig/URPM/Resolve.pm	2013-11-27 10:01:23.989148248 -0500
+++ URPM-4.48.2/URPM/Resolve.pm	2013-11-27 10:02:17.552142901 -0500
@@ -225,10 +225,12 @@
 	    $pkg->flag_requested > $p->flag_requested ||
 	      $pkg->flag_requested == $p->flag_requested && $pkg->compare_pkg($p) > 0 and $packages{$pkg->name} = $pkg;
 	} else {
-	    foreach my $altpkg ($urpm->packages_providing($pkg->name)) {
-	        if( URPM::rpmEVRcompare($altpkg->EVR, $pkg->EVR) > 0 ) {
- 		    $urpm->{debug_URPM}("Skipping ".$pkg->fullname." since newer version of the package exists in repositories" ) if $urpm->{debug_URPM};
-	    	    return;
+	    if ($pkg->fullname !~ /\.src$/) {
+	        foreach my $altpkg ($urpm->packages_providing($pkg->name)) {
+	            if( URPM::rpmEVRcompare($altpkg->EVR, $pkg->EVR) > 0 ) {
+ 		        $urpm->{debug_URPM}("Skipping ".$pkg->fullname." since newer version of the package exists in repositories" ) if $urpm->{debug_URPM};
+	    	        return;
+	            }
 	        }
 	    }
 	    $packages{$pkg->name} = $pkg;
